src/SpectatorView.Unity/Assets/SpectatorView/Scripts/Calibration/HeadsetCalibration.cs

```diff
@@ -243,5 +263,13 @@ private static MarkerCorners CalculateMarkerCorners(Vector3 topLeftPosition, Qua
             corners.orientation = topLeftOrientation;
             return corners;
         }
+
+#if UNITY_EDITOR
+        private void OnValidate()
+        {
+            FieldHelper.ValidateType<QRCodeMarkerDetector>(markerDetector);
+            FieldHelper.ValidateType<ArUcoMarkerDetector>(markerDetector);

```

Comment:
I think the preference here would be to log an error on Enable if this type is wrong. Right now these two checks will succeed if the marker detector is null, which we probably dont' want.